binchihua/test/common/parse-util_test.c

```diff
@@ -47,23 +50,31 @@ void test_parse_port(const char *input, bool expected_return, uint16_t expected_
 
         if (msg != NULL) {
                 fprintf(stderr, "%s", msg);
+                return false;
         }
+        return true;
 }
 
 int main() {
-        test_parse_long(NULL, false, 0L);
-        test_parse_long("", false, 0L);
-        test_parse_long("foo", false, 0L);
-        test_parse_long("1234", true, 1234L);
-        test_parse_long("1234abc", false, 1234L);
-        test_parse_long("abc1234", false, 0L);
+        bool result = true;
```

Comment:
Previously, we asserted the error, but did not return `EXIT_FAILURE` to give meson the signal that the test failed. Not sure if `result &&` is the best way to get the end result of all tests and return a failure if one case failed.

Reply:
Maybe it's just as well to have an assert at the first failure.

Reply:
For large tests such as the list test from systemd where every step (or test case) depends on the previous one this makes sense. In this case, each test case is independent and I'd really like to have a list of failures instead of getting one, fixing it, re-running, getting the next, etc. I'll leave it like this for now - we can still refactor later :)